### PR TITLE
NMS-8204: Reusing existing VMware session getting data

### PR DESCRIPTION
**Pull request by indigo423**
Allow reusing existing sessions to VMware by getting data instead of login/logout for every request.

* JIRA: http://issues.opennms.org/browse/NMS-8204

### CODE REVIEW

**Comment by mvrueden** [comment]
@christianpape can you please take a look.

File: integrations/opennms-vmware/src/main/java/org/opennms/protocols/vmware/VmwareViJavaAccess.java [hunk 1]
```diff
@@ -202,7 +210,26 @@ public VmwareViJavaAccess(String hostname) throws MarshalException, ValidationEx
     public void connect() throws MalformedURLException, RemoteException {
         relax();
 
-        m_serviceInstance = new ServiceInstance(new URL("https://" + m_hostname + "/sdk"), m_username, m_password);
+	//Fix for NMS-8204: use singleton of serverConnection, also make method synchronized
+	String connectionString = new String(m_hostname + m_username + m_password);
+	if(m_userSessions.get(connectionString) == null)
```
Comment (j-white) [inline integrations/opennms-vmware/src/main/java/org/opennms/protocols/vmware/VmwareViJavaAccess.java]: Are the sessions thread safe? What happens if two different threads try and the same session?

File: integrations/opennms-vmware/src/main/java/org/opennms/protocols/vmware/VmwareViJavaAccess.java [hunk 2]
```diff
@@ -233,19 +260,22 @@ public boolean setTimeout(int timeout) {
      * Disconnects from the server.
      */
     public void disconnect() {
-        if (m_serviceInstance == null) {
+	//Fix for NMS-8204: Doing nothing here
```
Comment (j-white) [inline integrations/opennms-vmware/src/main/java/org/opennms/protocols/vmware/VmwareViJavaAccess.java]: So we never close the sessions, and just wait for them to timeout?

File: integrations/opennms-vmware/src/main/java/org/opennms/protocols/vmware/VmwareViJavaAccess.java (hunk 1 above)
Comment (christianpape) [reply integrations/opennms-vmware/src/main/java/org/opennms/protocols/vmware/VmwareViJavaAccess.java]: Shouldn't this block/method be "synchronized"?

**Comment by christianpape** [comment]
This approach is simple, but we have to assure that this handling is thread-safe and stable concerning configuration changes during runtime. I don't know if we really need a more complicated connection pooling/management. In my point of view these connections will be opened and used/reopened as long as OpenNMS (data collection, provisioning, polling) is running. Stale connections will happen on configuration changes provisiond-configuration.xml or vmware-config.xml. I will try to run this branch using my two vCenter servers (v6.5 and v5.5).

**Comment by christianpape** [comment]
Yes, agree.

**Comment by christianpape** [comment]
Please see #2075

**Comment by j-white** [comment]
Closing in favor of https://github.com/OpenNMS/opennms/pull/2075.

**Pull request closed without merging**